detection_rules/rule.py

```diff
@@ -462,6 +460,61 @@ class ThresholdCardinality:
     threshold: ThresholdMapping
 
 
+@dataclass(frozen=True)
+class NewTermsRuleData(QueryRuleData):
+    """Specific fields for new terms field rule."""
+
+    @dataclass(frozen=True)
+    class NewTermsMapping(MarshmallowDataclassMixin):
+        @dataclass(frozen=True)
+        class HistoryWindowStart:
+            field: definitions.NonEmptyStr
+            value: definitions.HistoryWindowStart
+
+        field: definitions.NonEmptyStr
+        value: definitions.NewTermsFields
+        history_window_start: List[HistoryWindowStart]
```

Comment:
same here

```suggestion
        history_window_start: List[str]
```

In definitions, this was just a `str` - is that right or does it need to be a `NonEmptyStr`?

Reply:
`HistoryWindowStart` is the sub-class of NewTermsMapping. This does not pull from definitions.py and must stay.